modules/meteroid/crates/meteroid-store/src/domain/invoices.rs

```diff
@@ -262,6 +289,63 @@ impl InvoiceTotals {
             total,
             tax_amount,
             applied_credits,
+            applied_coupons: coupons_discount.applied_coupons,
+        }
+    }
+
+    fn calculate_coupons_discount(
+        subtotal: i64,
+        invoice_currency: &str,
+        coupons: &[AppliedCouponDetailed],
+    ) -> AppliedCouponsDiscount {
+        let applicable_coupons: Vec<&AppliedCouponDetailed> = coupons
+            .iter()
+            .filter(|x| x.is_invoice_applicable())
+            .sorted_by_key(|x| x.applied_coupon.created_at)
+            .collect::<Vec<_>>();
+
+        let mut applied_coupons: BTreeMap<Uuid, i64> = BTreeMap::new();
```

Comment:
so currently coupons are ordered by ids, right ? (that's fine in v1, it's just to understand)

Reply:
omg, I misused here BTreeMap, I will switch to either Vec or find a map that preserves the order

Reply:
fixed